### ISSUE TITLE
Directoy Diffing

### ISSUE THREAD

**Issue by rgrinberg** [sindrets/diffview.nvim#286]
Would it be possible to view diffs using this plugin outside the context of VCS? In particular, I'd like to diff two directories and browse the diff using the diffview UI.

**Comment by sindrets** [comment]
We could potentially allow passing the `--no-index` flag to `:DiffviewOpen` which would solve this:

```
       git diff [<options>] --no-index [--] <path> <path>
           This form is to compare the given two paths on the filesystem. [...]
```

But I believe this would require a fair amount of refactoring, as a lot of the git adapter code requires a reference to a git repository.

I'll consider it in the future.

**Comment by rgrinberg** [comment]
Right, that would be pretty neat. It would also be pretty good to navigate a diff loaded from a patch file. It's not such a common workflow anymore, but I do run into patches from time to time.

**Comment by sindrets** [comment]
The patch file idea is interesting. Feel free to open an issue about it so that we can track it separately.

**Comment by NeilGirdhar** [comment]
If you do support dir-diffs, it would be great to somehow preserve the configuration files that guide linters and text editing, like: .editorconfig and pyproject.toml.  Without these, linters are completely confused.

**Comment by strega-nil** [comment]
This feature would be a serious game changer for me. I absolutely love diffview, but I've been switching to jj recently and I just want diffview to work well with the default jj working setup :)

**Comment by dlyongemallo** [comment]
@rgrinberg @NeilGirdhar @strega-nil `:DiffviewDiffDirs` is implemented in [my fork](https://github.com/dlyongemallo/diffview.nvim). Please try it and let me know if it suffices for your needs.
